Documentation/storage-providers.md

```diff
@@ -0,0 +1,98 @@
+---
+title: Storage Providers
+weight: 12050
+indent: true
+---
+
+# Storage Providers
+
+Rook is the home for operators for multiple storage providers. Each of these storage providers
+has specific requirements and each of them is very independent. There is no runtime dependency
+between the storage providers. Development is where the storage providers benefit one another.
+
+Rook provides a development framework with a goal of enabling storage providers to create
+operators for Kubernetes to manage their storage layer. As the storage provider community
+grows, we expect this framework to grow as common storage constructs are identified
+that will benefit the community. Rook does not aim to replace other frameworks or
+communities, but to fill gaps not provided by other core projects. Here are
+some examples of other frameworks areas that already meet many community needs:
+
+* [Operator SDK](https://sdk.operatorframework.io/): For creating general operators.
+* [Storage SIG](https://github.com/kubernetes/community/tree/master/sig-storage):
+  For K8s storage API included in K8s, CSI drivers, snapshotting, local storage, and much more.
+* [Controller Runtime](https://github.com/kubernetes-sigs/controller-runtime):
+  Set of libraries for building Controllers for K8s.
+
```

Comment:
I do not get complete clarity whether these should be mutually exclusive with rook. If I am not mistaken, rook-ceps uses the controller-runtime (as of rather recently).  So using the runtime does not seem to contradict being part of rook.

So would it be thinkable for a provider to base its operator on the SDK, but still be part of the rook project, using some of the infra / libs and be part of the community?  Or is that undesirable by policy (or even technically impossible)?

I think the text could be a little clearer here.

Finally, a nit: I don't see the k8s storage sig as a framework but as the group responsible for the storage primitives in Kubernetes. 😃

Reply:
Just to add to that, if we want to be fair we would mention https://github.com/kubernetes-sigs/kubebuilder too, but since both (operator-sdk and kubebuilder) under the hood use controller-runtime perhaps we shouldn't mention any of them? Both or none?

Reply:
I'd say mention kubebuilder and sdk as frameworks, and maybe don't mention controller-runtime (or mention it as a library that e.g. rook-ceps is using). 🤔

Reply:
Agreed, will clarify this section. The controller runtime is currently used in Rook, and others like the operator sdk and kubebuild could easily be used by Rook operators in the future. The separate repos per storage provider will enable this such that we don't need to worry about conflicting choices across operators. The point is really to consider if they want to make the commitment to join the Rook community, as opposed to having their project completely independent.